### DISCUSSION TITLE
running 15 minute audio on pyannote/speaker-diarization@2.1

### DISCUSSION THREAD

**Discussion by khalid4294** [pyannote/pyannote-audio#1194] [Q&A]
Hey!
This model is very useful!
I'm encountering duration issue. I'm running this code on a 15 minute wave file takes too long and keeps running if I don't stop it.
I tried to run on shorter duration: 60 seconds - and that took around 6.5 minutes.

```
from pyannote.audio import Pipeline
pipeline = Pipeline.from_pretrained("pyannote/speaker-diarization@2.1", use_auth_token=h_token)
output = pipeline("audio/mini_cut.wav", num_speakers=2)
```

I just want to make sure that the time it takes is normal and it's not something related to my machine. 
I'm using mac m1 chip, code running on a Jupyter notebook extension on VScode.

Can anyone confirm? is there a faster way?

**Comment by k3ybladewielder** [comment]
How long, @khalid4294 ?
For the duration, it would take approximately 15 to 20 minutes

**Reply by khalid4294** [reply]
the audio file I have is 15min long conversation between 2 speakers.
